api/registrations/serializers.py

```diff
@@ -275,16 +276,35 @@ def get_current_user_permissions(self, obj):
         return NodeSerializer.get_current_user_permissions(self, obj)
 
     def update(self, registration, validated_data):
-        is_public = validated_data.get('is_public', False)
-        if is_public:
-            try:
-                registration.update(validated_data)
-            except NodeUpdateError as err:
-                raise exceptions.ValidationError(err.reason)
-            except NodeStateError as err:
-                raise exceptions.ValidationError(err.message)
+        #Update tags
+        auth = get_user_auth(self.context['request'])
+        old_tags = set(registration.tags.values_list('name', flat=True))
+        if 'tags' in validated_data:
+            current_tags = set(validated_data.pop('tags', []))
+        elif self.partial:
+            current_tags = set(old_tags)
         else:
-            raise exceptions.ValidationError('Registrations can only be turned from private to public.')
+            current_tags = set()
+
+        try:
+            for new_tag in (current_tags - old_tags):
```

Comment:
Since this is essentially the same as what's in the node model, it might be good to DRY it up a bit.